Detectors/MUON/MCH/Align/src/AlignmentSpec.cxx

```diff
@@ -0,0 +1,949 @@
+        base::Propagator::initFieldFromGRP(grp);
+        TrackExtrap::setField();
+        mAlign.SetBFieldOn(TrackExtrap::isFieldON());
+        TrackExtrap::useExtrapV2();
+      } else {
+        LOG(fatal) << "No GRP file";
+      }
+
+      auto geoIdealFile = ic.options().get<string>("geo-file-ideal");
+      if (std::filesystem::exists(geoIdealFile)) {
+        base::GeometryManager::loadGeometry(geoIdealFile.c_str());
+        transformation = geo::transformationFromTGeoManager(*gGeoManager);
+        for (int i = 0; i < 156; i++) {
+          int iDEN = GetDetElemId(i);
+          transformIdeal[iDEN] = transformation(iDEN);
+        }
+      } else {
+        LOG(fatal) << "No ideal geometry";
+      }
+
+      auto geoRefFile = ic.options().get<string>("geo-file-ref");
+      if (std::filesystem::exists(geoRefFile)) {
+        base::GeometryManager::loadGeometry(geoRefFile.c_str());
+        transformation = geo::transformationFromTGeoManager(*gGeoManager);
+        for (int i = 0; i < 156; i++) {
+          int iDEN = GetDetElemId(i);
+          transformRef[iDEN] = transformation(iDEN);
+        }
+      } else {
+        LOG(fatal) << "No reference geometry";
+      }
+    }
+
+    trackFitter.smoothTracks(true);
+    trackFitter.setChamberResolution(Reso_X, Reso_Y);
+    trackFitter.useChamberResolution();
+
+    mAlign.SetDoEvaluation(true);
+    // Variation range for parameters
+    mAlign.SetAllowedVariation(0, 2.0);
+    mAlign.SetAllowedVariation(1, 0.3);
+    mAlign.SetAllowedVariation(2, 0.002);
+    mAlign.SetAllowedVariation(3, 2.0);
+
+    // Fix chambers
+    auto chambers = ic.options().get<string>("fix-chamber");
+    for (int i = 0; i < chambers.length(); ++i) {
+      if (chambers[i] == ',') {
+        continue;
+      }
+      int chamber = chambers[i] - '0';
+      LOG(info) << Form("%s%d", "Fixing chamber: ", chamber);
+      mAlign.FixChamber(chamber);
+    }
+
+    doMatched = ic.options().get<bool>("matched");
+    outFileName = ic.options().get<string>("output");
+    readFromRec = ic.options().get<bool>("use-record");
+
+    if (readFromRec) {
+      mAlign.SetReadOnly();
+      LOG(info) << "Reading records as input";
+    }
+    mAlign.init();
+
+    ic.services().get<CallbackService>().set<CallbackService::Id::Stop>([this]() {
+      LOG(info) << "Alignment duration = " << mElapsedTime.count() << " s";
+    });
+  }
+
+  //_________________________________________________________________________________________________
+  void finaliseCCDB(framework::ConcreteDataMatcher& matcher, void* obj)
+  {
+    /// finalize the track extrapolation setting
+    if (mCCDBRequest && base::GRPGeomHelper::instance().finaliseCCDB(matcher, obj)) {
+
+      if (matcher == framework::ConcreteDataMatcher("GLO", "GRPMAGFIELD", 0)) {
+        LOG(info) << "Loading magnetic field from CCDB";
+        auto grp = base::GRPGeomHelper::instance().getGRPMagField();
+        base::Propagator::initFieldFromGRP(grp);
```

Comment:
you can remove these 2 lines above since the field is already set in `GRPGeomHelper::finaliseCCDB(...)`

Reply:
Indeed, thanks a lot for pointing it out.